examples/protonect/src/depth_packet_stream_parser.cpp

```diff
@@ -59,122 +54,86 @@ void DepthPacketStreamParser::setPacketProcessor(libfreenect2::BaseDepthPacketPr
 
 void DepthPacketStreamParser::onDataReceived(unsigned char* buffer, size_t in_length)
 {
-  // TODO: simplify this crap (so code, such unreadable, wow ;)
-  Buffer &wb = work_buffer_;
+  if (in_length == 0)
+    return;
 
-  size_t in_offset = 0;
+  DepthSubPacketFooter *footer = 0;
 
-  while(in_offset < in_length)
-  {
-    unsigned char *ptr_in = buffer + in_offset, *ptr_out = wb.data + wb.length;
-    DepthSubPacketFooter *footer = 0;
-    bool footer_found = false;
+  Buffer &fb = buffer_.front();
 
-    size_t max_length = std::min<size_t>(wb.capacity - wb.length, in_length - 8);
+  for (size_t i = 0; i < in_length; i++)
```

Comment:
upper bound should be `in_length - sizeof(DepthSubPacketFooter)` otherwise we might cause a segfault!?